tests/driver/index.ts

```diff
@@ -0,0 +1,322 @@
+  };
+
+  /**
+   * Parse the result of 'getFullTrunkConfig' in the context of 'ARGS' to identify the desired linter version to enable.
+   */
+  extractLinterVersion = (): string => {
+    if (!ARGS.linterVersion || ARGS.linterVersion === "Latest") {
+      return "";
+    } else if (ARGS.linterVersion === "KnownGoodVersion") {
+      // trunk-ignore-begin(eslint/@typescript-eslint/no-unsafe-member-access,eslint/@typescript-eslint/no-unsafe-call)
+      return (
+        (this.getFullTrunkConfig().lint.definitions.find(
+          ({ name }: { name: string }) => name === this.linter
+        )?.known_good_version as string) ?? ""
+      );
+      // trunk-ignore-end(eslint/@typescript-eslint/no-unsafe-member-access,eslint/@typescript-eslint/no-unsafe-call)
+    } else {
+      return ARGS.linterVersion;
+    }
+  };
+
+  /**
+   * Setup a sandbox test directory by copying in test contents and conditionally:
+   * 1. Creating a git repo
+   * 2. Dumping a newly generated trunk.yaml
+   * 3. Launching the daemon
+   * 4. Enabling the specified 'linter'
+   */
+  async setUp() {
+    this.sandboxPath = fs.mkdtempSync(path.resolve(os.tmpdir(), TEMP_PREFIX));
+    if (!this.sandboxPath) {
+      return;
+    }
+    // Create repo
+    const sourceDir = path.resolve(this.testDir, "..");
+    fs.cpSync(sourceDir, this.sandboxPath, { recursive: true });
+
+    if (this.setupSettings.setupGit) {
+      this.gitDriver = git.simpleGit(this.sandboxPath);
+      await this.gitDriver
+        .init()
+        .add(".")
+        .addConfig("user.name", "Plugin Author")
+        .addConfig("user.email", "trunk-plugins@example.com")
+        .commit("first commit");
+    }
+
+    if (this.setupSettings.setupTrunk) {
+      // Initialize trunk via config
+      fs.mkdirSync(path.resolve(this.sandboxPath, ".trunk"));
+      fs.writeFileSync(path.resolve(this.sandboxPath, ".trunk/trunk.yaml"), newTrunkYamlContents());
+    }
+
+    // Launch daemon if specified
+    if (!this.setupSettings.launchDaemon) {
+      return;
+    }
+
+    // Prefer calling enable over editing trunk.yaml directly because it also handles runtimes, etc.
+    const trunkCommand = ARGS.cliPath ?? "trunk";
+    const daemonArgs = ["daemon", "launch", "--monitor=false"];
+
+    this.daemon = execFile(trunkCommand, daemonArgs, {
+      cwd: this.sandboxPath,
+    });
+
+    // Enable tested linter if specified
+    if (!this.linter) {
+      return;
+    }
+
+    try {
+      const version = this.extractLinterVersion();
+      const versionString = version.length > 0 ? `@${version}` : "";
+      const linterVersionString = `${versionString}${this.linter}`;
+      // Prefer calling `check enable` over editing trunk.yaml directly because it also handles runtimes, etc.
+      await this.run(`check enable ${linterVersionString} --monitor=false`);
+    } catch (error) {
+      console.warn(`Failed to enable ${this.linter} with message ${(error as Error).message}`);
+      console.warn(error);
```

Comment:
nit, totally optional: these logs seem a little duplicative since the `error.message` will be printed normally

```suggestion
      console.warn(`Failed to enable ${this.linter}`, error);
```